string source
string id # these ids should be unique (ideally globally but at least w.r.t source)

string level
string LEVEL_NOTE=note
string LEVEL_WARN=warn
string LEVEL_ERROR=error

string data
string human_message
